### PR TITLE
bpo-29271: Fix Task.current_task and Task.all_tasks to accept None.

### PR DESCRIPTION
**Pull request by 1st1**
See http://bugs.python.org/issue29271 for context. LGTM'ed by @methane

### CODE REVIEW

File: Misc/NEWS [hunk 1]
```diff
@@ -688,6 +688,9 @@ Library
 - Issue #24142: Reading a corrupt config file left configparser in an
   invalid state.  Original patch by Florian Höch.
 
+- Issue #29271: Fix Task.current_task and Task.all_tasks to accept
```
Comment (serhiy-storchaka) [inline Misc/NEWS]: Move the entry to the start of the section. Mention that this fixes C implementation, makes it conforming to Python implementation.

File: Lib/test/test_asyncio/test_tasks.py [hunk 2]
```diff
@@ -1461,6 +1461,14 @@ def test_current_task(self):
         def coro(loop):
             self.assertTrue(Task.current_task(loop=loop) is task)
 
+            # See http://bugs.python.org/issue29271 for details:
+            asyncio.set_event_loop(loop)
+            try:
+                self.assertTrue(Task.current_task(None) is task)
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_tasks.py]: Why not use `assertIs()`?

Comment (1st1) [reply Lib/test/test_asyncio/test_tasks.py]: good catch

File: Misc/NEWS (hunk 1 above)
Comment (1st1) [reply Misc/NEWS]: Aren't we usually appending news items instead of prepending?

File: Misc/NEWS [hunk 3]
```diff
@@ -256,6 +256,9 @@ Extension Modules
 Library
 -------
 
+- Issue #29271: Fix Task.current_task and Task.all_tasks implemented in C
```
Comment (serhiy-storchaka) [inline Misc/NEWS]: Now Misc/NEWS entries should be prefixed by "bpo-XXXXX" rather of "Issue #XXXXX".

**Comment by serhiy-storchaka** [comment]
> Aren't we usually appending news items instead of prepending?

Entries in Misc/NEWS should be prepended. Entries in What's New usually are 
appended.

**Comment by 1st1** [comment]
Thank you for helping out Serhiy. It's my first PRs to the new GH repo so I'm getting accustomed to some new practices.

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by 1st1** [comment]
@serhiy-storchaka BTW, to backport this to 3.6, do I need to create a new PR? Or I can just cherry-pick and push?

**Comment by serhiy-storchaka** [comment]
AFAIK you can't push directly into this repository. You need to create a new 
PR. Add the label "need to backport to 3.6" to this PR and "cherry-pick for 
3.6" to the new PR.

**Pull request merged**